core/src/test/java/org/apache/calcite/test/RelMetadataTest.java

```diff
@@ -1438,6 +1438,23 @@ private void checkPredicates(RelOptCluster cluster, RelOptTable empTable,
     assertThat(pulledUpPredicates, sortsAs("[=($0, 1)]"));
   }
 
+  @Test public void testPullUpPredicatesForExprsItr() {
+    final String sql = "select a.EMPNO, a.ENAME from (select * from sales.emp )"
+        + " a join (select * from sales.emp  ) b on a.empno = b.deptno and"
+        + " a.comm = b.comm and a.mgr=b.mgr and (a.empno < 10 or a.comm < 3 "
+        + "or a.deptno < 10 or a.job ='abc' or a.ename='abc' or a.sal='30' or "
```

Comment:
Hi @LeoWangLZ , I'm learning this PR ~
In espisode of `a.sal='30'`, you are comparing a numeric with characters, did you really mean this ?